### PR TITLE
Quote markdown tweaks

### PR DESCRIPTION
**Pull request by mislav**
- Preserve ~strikethrough~ when quoting markdown
- "Suggested Changes" blocks are omitted when quoting markdown
- Blob snippets are quoted simply as the original blob link that generated the snippet

### CODE REVIEW

File: markdown-parsing.js [hunk 1]
```diff
@@ -180,6 +183,22 @@ const filters: {[key: string]: (HTMLElement) => string | HTMLElement} = {
   },
   UL(el) {
     return el
+  },
+  DIV(el) {
+    if (el.classList.contains('js-suggested-changes-blob')) {
```
Comment (koddsson) [inline markdown-parsing.js]: Could this somehow live in the application code since it seems tightly coupled with it?

File: markdown-parsing.js [hunk 2]
```diff
@@ -180,6 +183,22 @@ const filters: {[key: string]: (HTMLElement) => string | HTMLElement} = {
   },
   UL(el) {
     return el
+  },
+  DIV(el) {
```
Comment (muan) [inline markdown-parsing.js]: Is this something we can do in dotcom by catching the `quote-selection` event? These are very dotcom specific and would be nice to not include in this module.

**Comment by mislav** [comment]
@koddsson @muan Excellent point about GitHub-specific syntax. I've removed all GitHub-specific markdown features from this library and published a new event `quote-selection-markdown` that serves as a hook through which we can implement more syntaxes.

The existing `quote-selection` event fires too late to be able to implement this. Before creating a new event, I've considered firing `quote-selection` a bit earlier and exposing an additional object in its `details`, but since the selection-to-markdown functionality is conditional (only kicks in if `data-quote-markdown` attribute is present), it feels a bit awkward to try to move event firing to that phase but also ensuring that it fires in the non-markdown mode.

Do you have another idea how these events could be organized?

**Comment by mislav** [comment]
I've pushed another change to get rid of reading the `data-quote-markdown` attribute. Now, opting into markdown mode can be done via optional parameters to `install/subscribe()`! This means even fewer GitHub-specific conventions needed to set up this library 🌟 Feedback on API welcome!

**Comment by mislav** [comment]
I've pushed another change to get rid of reading the `data-quote-markdown` attribute. Now, opting into markdown mode can be done via optional parameters to `install/subscribe()`! This means even fewer GitHub-specific conventions needed to set up this library 🌟 Feedback on API welcome!

**Comment by mislav** [comment]
I've pushed another change to get rid of reading the `data-quote-markdown` attribute. Now, opting into markdown mode can be done via optional parameters to `install/subscribe()`! This means even fewer GitHub-specific conventions needed to set up this library 🌟 Feedback on API welcome!

**Comment by mislav** [comment]
I've pushed another change to get rid of reading the `data-quote-markdown` attribute. Now, opting into markdown mode can be done via optional parameters to `install/subscribe()`! This means even fewer GitHub-specific conventions needed to set up this library 🌟 Feedback on API welcome!

File: test/test.js [hunk 3]
```diff
@@ -132,19 +126,34 @@ describe('quote-selection', function() {
...
 
 `
       )
     })
+
+    it('allows quote-selection-markdown event to prepare content', function() {
+      document.querySelector('[data-quote]').addEventListener('quote-selection-markdown', function(event) {
+        const {fragment} = event.detail
+        fragment.querySelector('a[href]').replaceWith('@links')
+        fragment.querySelector('img[alt]').replaceWith(':emoji:')
```
Comment (muan) [inline test/test.js]: 💯 !

**Review by muan — APPROVED** [review]
I like this a lot! I didn't realized so many app specific behaviors were baked into this. 😓 

It seems like the language syntax check might be a bit tricky to synchronize on the application side. But regardless I think it's a worthwhile change. 👍

**Pull request merged**